File: api/neck/Models/Entity/Instrument.cs
using neck.Interfaces;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace neck.Models.Entity
{
    public class Instrument : DbEntity, ILabelled
    {
        public string Label { get; set; }

        public int NumStrings { get; set; }

        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public Guid? DefaultTuningId { get; set; }

        public Tuning DefaultTuning { get; set; }

        [JsonIgnore]
        public virtual ICollection<Tuning> Tunings { get; set; }
    }
}

File: web/src/models/api/Instrument.ts
import { ApiEntity } from '../../network/ApiEntity';
import { Formation } from './Formation';
import { Tuning } from './Tuning';

export class Instrument extends ApiEntity {
  public DefaultTuning: Tuning;
  public DefaultTuningId: string;
  public Label: string;
  public NumFrets: number;
  public NumStrings: number;
  public Formation: Formation;
}
